src/Resolvers/Microsoft.DotNet.NativeWrapper/Interop.cs

```diff
@@ -12,13 +13,23 @@ namespace Microsoft.DotNet.NativeWrapper
     public static partial class Interop
     {
         public static readonly bool RunningOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+#if NETCOREAPP
+        private static readonly string HostFxrPath;
+#endif
 
         static Interop()
         {
             if (RunningOnWindows)
             {
-                PreloadWindowsLibrary("hostfxr.dll");
+                PreloadWindowsLibrary(Constants.HostFxr);
             }
+#if NETCOREAPP
+            else
+            {
+                HostFxrPath = (string)AppContext.GetData(Constants.RuntimeProperty.HostFxrPath);
+                System.Runtime.Loader.AssemblyLoadContext.Default.ResolvingUnmanagedDll += HostFxrResolver;
```

Comment:
I think this should be on the current ALC instead - `AssemblyLoadContext.GetLoadContext(Assembly.GetExecutingAssembly())`.